code/game/machinery/doors/firedoor.dm

```diff
@@ -361,6 +367,15 @@ var/global/list/alert_overlays_global = list()
 
 	do_interaction(user, C)
 
+/obj/machinery/door/firedoor/emag_ai(mob/living/silicon/ai/A)
+	if(density)
+		flick("door_spark", src)
+		sleep(6)
+		open()
+		sleep(8)
```

Comment:
odd behavior where if you reclose it quick enough it becomes welded closed instead of welded open? intended?

Reply:
This is how regular emags do it, I've never seen that happen but I'll remove the sleep(8)